### PR TITLE
BUGFIX: SD-825 Attribute name in tooltip isn't updated after renaming

### PR DESCRIPTION
**Pull request by dlcd2203**
---

**Proposed release type:** patch

# PR checklist

- [x] Change was tested using dev-release in Analytical Designer and Dashboards (if applicable).
- [ ] Change is described in [CHANGELOG.md](../blob/master/CHANGELOG.md).
- [ ] Migration guide (for major update) is written to [CHANGELOG.md](../blob/master/CHANGELOG.md).
- [ ] The proposed release type is appropriate (see the comment in [PR template](../blob/master/.github/pull_request_template.md))

# Related PRs
- gdc-analytical-designer: https://github.com/gooddata/gdc-analytical-designer/pull/2874
- gdc-dashboards: https://github.com/gooddata/gdc-dashboards/pull/2613

# Related Jira tasks
- SD-825: https://jira.intgdc.com/browse/SD-825

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-js-oss-check/83/ : SUCCESS in 5s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-js-unit-tests-zuul-docker-el7/517/ : SUCCESS in 1m 56s

**Comment by no23reason** [comment]
Hi, please specify the proposed release version of this change (see the PR template).

File: src/DataLayer/converters/toAfmResultSpec.ts [hunk 1]
```diff
@@ -130,7 +131,8 @@ function getGeoAttributeForTooltip(
     const properties: IVisualizationProperties = parsePropertyItem(visualizationObject.properties || "");
     const tooltipText: string = get(properties, "controls.tooltipText", "");
     if (tooltipText) {
-        return buildTooltipBucketItem(tooltipText);
+        const alias: string = get(visualizationObject, "buckets[0].items[0].visualizationAttribute.alias");
```
Comment (lvkhanh) [inline src/DataLayer/converters/toAfmResultSpec.ts]: I suggest we should get bucket location by localIdentifier instead of get first bucket item

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-js-oss-check/84/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-js-unit-tests-zuul-docker-el7/518/ : SUCCESS in 57s

File: src/DataLayer/converters/toAfmResultSpec.ts [hunk 2]
```diff
@@ -130,8 +131,15 @@ function getGeoAttributeForTooltip(
     const properties: IVisualizationProperties = parsePropertyItem(visualizationObject.properties || "");
     const tooltipText: string = get(properties, "controls.tooltipText", "");
     if (tooltipText) {
-        return buildTooltipBucketItem(tooltipText);
+        const locationBucket: VisualizationObject.IBucket | undefined = visualizationObject.buckets.find(
+            (obj: VisualizationObject.IBucket): boolean => {
```
[the remark this answers is not in the document]
Comment (naibinh) [reply src/DataLayer/converters/toAfmResultSpec.ts]: and how about this
```
const locationAlias = visualizationObject.buckets.reduce((locationAlias: string, bucket: VisualizationObject.IBucket): string => {
    if (!locationAlias && bucket.localIdentifier === LOCATION) {
        return get(bucket, "items[0].visualizationAttribute.alias");
    }
    return locationAlias;
}, undefined);
```

File: src/DataLayer/converters/toAfmResultSpec.ts (hunk 1 above)
Comment (dlcd2203) [reply src/DataLayer/converters/toAfmResultSpec.ts]: fixed

File: src/DataLayer/converters/toAfmResultSpec.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (dlcd2203) [reply src/DataLayer/converters/toAfmResultSpec.ts]: fixed

File: src/DataLayer/converters/toAfmResultSpec.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (dlcd2203) [reply src/DataLayer/converters/toAfmResultSpec.ts]: fixed

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-js-oss-check/85/ : SUCCESS in 5s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-js-unit-tests-zuul-docker-el7/519/ : SUCCESS in 2m 11s

**Comment by lvkhanh** [comment]
LGTM

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-js-oss-check/86/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-js-unit-tests-zuul-docker-el7/520/ : SUCCESS in 1m 11s

**Comment by naibinh** [comment]
LGTM

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-js-merge-compliance-check/120/ : SUCCESS in 24s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-js-unit-tests-zuul-docker-el7/521/ : SUCCESS in 2m 27s

**Pull request merged**